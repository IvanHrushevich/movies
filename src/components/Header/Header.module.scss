@import '../../shared/variables.scss';

.container {
  padding: 1rem $container-padding 0 $container-padding;
  height: 40rem;
  display: flex;
  flex-direction: column;
  background-color: $header-color;
}

.searchBlock {
  margin: 6rem 6rem 0 11rem;
}

.title {
  font-size: 4rem;
  margin-bottom: 3rem;
}

.sortPanel {
  margin-top: auto;
}
