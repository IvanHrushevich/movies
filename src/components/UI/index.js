export * from './Button/Button';
export * from './SortController/SortController';
