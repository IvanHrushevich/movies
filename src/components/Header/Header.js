import React from 'react';
import { connect } from 'react-redux';

import classes from './Header.module.scss';
import SortPanel from '../SortPanel/SortPanel';
import { Logo, SearchControl, SortControl } from '../UI/index';
import { movieActions } from '../../store/index';

export const Header = (props) => (
  <>
    <header className={classes.sectionHeader}>
      <Logo />
      <div className={`container-global ${classes.searchBlock}`}>
        <p className={classes.title}>FIND YOUR MOVIE</p>
        <div className={classes.searchControl}>
          <SearchControl />
        </div>
        <SortControl
          title="search by"
          btn1={{
            label: 'title',
            value: 'title',
            checked: true,
          }}
          btn2={{
            label: 'genre',
            value: 'genre',
          }}
          inputName="searchBy"
          onChange={(event) => props.changeSearchBy(event.target.value)}
        />
      </div>
    </header>
    <SortPanel />
  </>
);

const mapDispatchToProps = (dispatch) => ({
  changeSearchBy: (value) => dispatch(movieActions.changeSearchBy(value)),
});

export default connect(null, mapDispatchToProps)(Header);
