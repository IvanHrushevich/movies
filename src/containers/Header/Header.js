import React from 'react';
import { connect } from 'react-redux';

import classes from './Header.module.scss';
import SortPanel from '../SortPanel/SortPanel';
import { Logo, SearchControl, SortControl } from '../../components/index';
import { movieActions } from '../../store/index';

export const Header = (props) => (
  <>
    <header className={classes.sectionHeader}>
      <Logo />
      <div className={`container-global ${classes.searchBlock}`}>
        <p className={classes.title}>FIND YOUR MOVIE</p>
        <div className={classes.searchControl}>
          <SearchControl
            onSearchClick={(searchStr) => {
              props.changeSearchStr(searchStr);
              props.fetchMovies();
            }}
          />
        </div>
        <SortControl
          title="search by"
          btn1={{
            label: 'title',
            value: 'title',
            checked: true,
          }}
          btn2={{
            label: 'genre',
            value: 'genres',
          }}
          inputName="searchBy"
          onChange={(event) => {
            props.changeSearchBy(event.target.value);
            props.fetchMovies();
          }}
        />
      </div>
    </header>
    <SortPanel />
  </>
);

const mapDispatchToProps = (dispatch) => ({
  changeSearchStr: (value) => dispatch(movieActions.changeSearchStr(value)),
  changeSearchBy: (value) => dispatch(movieActions.changeSearchBy(value)),
  fetchMovies: () => dispatch(movieActions.fetchMovies()),
});

export default connect(null, mapDispatchToProps)(Header);
