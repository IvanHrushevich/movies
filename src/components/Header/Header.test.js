import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';
import { Logo, SearchControl, SortControl } from '../UI/index';

describe('Header', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Header />);
  });

  it('should render properly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should contain Logo, SearchControl and SortControl components', () => {
    const component = shallow(<Header />);

    expect(component.find(SearchControl)).toHaveLength(1);
    expect(component.find(SortControl)).toHaveLength(1);
    expect(component.find(Logo)).toHaveLength(1);
  });
});
