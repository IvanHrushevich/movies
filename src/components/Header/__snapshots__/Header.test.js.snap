// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<Fragment>
  <header
    className="sectionHeader"
  >
    <Logo />
    <div
      className="container-global searchBlock"
    >
      <p
        className="title"
      >
        FIND YOUR MOVIE
      </p>
      <div
        className="searchControl"
      >
        <SearchControl />
      </div>
      <SortControl
        btn1Label="title"
        btn2Label="gengre"
        inputName="searchBy"
        title="search by"
      />
    </div>
  </header>
  <SortPanel />
</Fragment>
`;
